### ISSUE TITLE
Websocket not working on Chrome iOS

### ISSUE THREAD

**Issue by jidma** [jhipster/generator-jhipster#11084]
##### **Overview of the issue**

Websocket not working on Chrome iOS

I get the following error on "chrome://inspect" :

```
https://xxxxxxxxx.com/admin/jhi-tracker
LOG Application started
LOG Opening Web Socket...
ERROR ERROR SecurityError: The operation is insecure.
LOG Whoops! Lost connection to https://xxxxxxxxx.com/websocket/tracker?access_token=eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImF1dGgiOiJST0xFX0FETUlOLFJPTEVfVVNFUiIsImV4cCI6MTU4MDg0ODcwMn0.wkyqSSQAFC5l0VJ3L0kqn0Y4jKjGdDUdAdkCnrK6Qmb1FLC7NrZtHBmm6B7_Pujg0R4f1Ug1E3C9igc6TGcyXQ: [object Object]
```

At first I thought it was due to SSL but even after securing my domain it still doesn't work.

It works fine on Chrome desktop (Ubuntu) and I can see the user activity when connected as Admin.

##### **Reproduce the error**

jhipster v6.2.0

Angular front end

Websocker enabled.

Navigate to the webapp and login as user in Chrome for iPhone.

##### **JHipster Version(s)**

jhipster v6.2.0

##### **JHipster configuration**

No relevant changes in the configuration.

##### **Browsers and Operating System**

iPhone X, Chrome

-   [X] Checking this box is mandatory (this is just to show you read everything)

**Comment by ruddell** [comment]
I reproduced it, it also seems to affect Safari on Mac, Chrome on iOS uses Webkit for rendering so it makes sense all Safari browsers are affected.

Error looks like something to do with the CSP config:
```
Refused to connect to wss://jhipster-chat-v-6.herokuapp.com/websocket/tracker/618/eeorglh4/websocket 
because it appears in neither the connect-src directive
nor the default-src directive of the Content Security Policy.
```

If you configure the CSP settings in [SecurityConfiguration](https://github.com/jhipster/jhipster-sample-app/blob/3d3ac52b397e8f14b7dd78c52b08a8833e3c5da7/src/main/java/io/github/jhipster/sample/config/SecurityConfiguration.java#L69) to include your domain (ie `ws://yourdomain.com`) in the `default-src` it works as expected.  It is weird that `self` doesn't work but it is using `ws://` instead of `http://`.

With the latest JHipster, the websocket falls back to using XHR requests which works fine (but is still not ideal).  The websocket fallback methods were fixed in https://github.com/jhipster/generator-jhipster/pull/10919
